File: FoodForecastingAPI/FoodForecastingAPI/Models/Meal.cs
namespace FoodForecastingAPI.Models
{
    public class Meal
    {
        public int Id { get; set; }

        public string MealName { get; set; } = "";

        public string MealType { get; set; } = "";

        public int Quantity { get; set; }

        public string Status { get; set; } = "Available";
    }
}

File: food-forecasting-ui/src/app/services/meal.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Meal {
  id?: number;
  mealName: string;
  mealType: string;
  quantity: number;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class MealService {

  private apiUrl = 'http://localhost:5003/api/Meals';

  constructor(private http: HttpClient) {}

  getMeals(): Observable<Meal[]> {
    return this.http.get<Meal[]>(this.apiUrl);
  }

  addMeal(meal: Meal): Observable<Meal> {
    return this.http.post<Meal>(this.apiUrl, meal);
  }

  deleteMeal(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

}